src/pdfviewer/PDFDocument.cpp

```diff
@@ -1586,9 +1589,16 @@ void PDFWidget::contextMenuEvent(QContextMenuEvent *event)
 		usingTool = kNone;
 	}
 
-	if (pdfDoc && pdfDoc->menuShow) {
-		menu.addSeparator();
-		menu.addMenu(pdfDoc->menuShow);
+	if (pdfDoc) {
+		if (pdfDoc->menuGrid || pdfDoc->menuShow) {
+			menu.addSeparator();
+			if (pdfDoc->menuGrid) {
+				menu.addMenu(pdfDoc->menuGrid);
+			}
+			if (pdfDoc->menuShow) {
+				menu.addMenu(pdfDoc->menuShow);
+			}
+		}
```

Comment:
overly complicated code
apart from the fact the the pointers are not zeroed in the first place, there is no case when only one of the two appears.